### PR TITLE
Update purpose.mdx

### PR DESCRIPTION
**Pull request by Mukesh05**
The document is less intuitive for beginners . I've added contents to make the document more lucid.

### CODE REVIEW

File: website/docs/language/state/purpose.mdx
Comment (laurapacilio) [inline website/docs/language/state/purpose.mdx:27]:
```suggestion
world. For example, when you have a resource `resource "aws_instance" "foo"` in your
configuration, Terraform uses this mapping to know that the resource `resource "aws_instance" "foo"`
represents a real world object with the instance ID `i-abcd1234` on a remote system.
```

File: website/docs/language/state/purpose.mdx
Comment (laurapacilio) [inline website/docs/language/state/purpose.mdx:70]:
```suggestion
ordering semantics of every resource for every _provider_, Terraform must also
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
@Mukesh05 Thank you! I left a couple of suggestions for you to review. 

@crw I would also like someone on engineering to double-check this to make sure I didn't miss anything. This should be a very quick review, but state is such an important concept that I'd like engineering to okay these edits as well.

File: website/docs/language/state/purpose.mdx
Comment (crw) [inline website/docs/language/state/purpose.mdx:38]: "on the provider" is ambiguous here. We could change this to "on the service provider," but I think it can be removed entirely -- "remote object" is clear enough.

File: website/docs/language/state/purpose.mdx
Comment (crw) [inline website/docs/language/state/purpose.mdx:39]: "guaranteed by" reads more clearly to me than "guaranteed since." I would not change this clause.

**Review by crw — CHANGES_REQUESTED** [review]
I added my comments. Overall, these are positive changes, @laurapacilio I give them my technical stamp of approval pending the corrections requested.

File: website/docs/language/state/purpose.mdx
Comment (Mukesh05) [reply website/docs/language/state/purpose.mdx:27]: @laurapacilio - "with the" looks more appropriate. Thank you.

File: website/docs/language/state/purpose.mdx
Comment (Mukesh05) [reply website/docs/language/state/purpose.mdx:39]: @crw  - Thank you. Could we change this to - "which is normally guaranteed by Terraform since Terraform is responsible for"

Comment (crw) [reply website/docs/language/state/purpose.mdx]: I think that is OK. @laurapacilio, please let us know if that sounds grammatically correct.

Comment (laurapacilio) [reply website/docs/language/state/purpose.mdx]: Hi both - apologies for the delay here! I want to think about this sentence in context (also I don't love how the original text separates out lines that are clearly related to one another into separate paragraphs). So I pulled out some additional sentences that come after this and rewrote all of them a bit. The goal is to make sure it all flows together. What do you think of rewriting it to be something like what I have below?

"Terraform expects that each remote object is bound to only one resource instance in the configuration. If a remote object is bound to multiple resource instances, the mapping from configuration to the remote object state becomes ambiguous, and Terraform may behave unexpectedly. Terraform can guarantee a one-to-one mapping when it creates objects and records their identities in the state. When importing objects created outside of Terraform, you must make sure that each distinct object is imported to only one resource instance."

Comment (Mukesh05) [reply website/docs/language/state/purpose.mdx]: Hi @laurapacilio - The paragraph looks more comprehensive. A minor correction in the paragraph is needed, though.
@crw  - please suggest.
"Terraform expects that each remote object is bound to only one resource instance in the configuration. If a remote object is bound to multiple resource instances, the mapping from configuration to the remote object in the state becomes ambiguous, and Terraform may behave unexpectedly. Terraform can guarantee a one-to-one mapping when it creates objects and records their identities in the state. When importing objects created outside of Terraform, you must make sure that each distinct object is imported to only one resource instance."

Comment (crw) [reply website/docs/language/state/purpose.mdx]: That minor change looks good (took me a while to find it).

Comment (Mukesh05) [reply website/docs/language/state/purpose.mdx]: Thank you @crw . 
@laurapacilio - All good . Below is the copy of your paragraph which includes my change that @crw  has consented to.
"Terraform expects that each remote object is bound to only one resource instance in the configuration. If a remote object is bound to multiple resource instances, the mapping from configuration to the remote object in the state becomes ambiguous, and Terraform may behave unexpectedly. Terraform can guarantee a one-to-one mapping when it creates objects and records their identities in the state. When importing objects created outside of Terraform, you must make sure that each distinct object is imported to only one resource instance."

Comment (laurapacilio) [reply website/docs/language/state/purpose.mdx]: This looks great! Could you please update the document to add this new, combined paragraph?  Once you do that, I feel comfortable approving and merging :)

Comment (Mukesh05) [reply website/docs/language/state/purpose.mdx]: @laurapacilio  - I have updated the existing document with the new, combined paragraph.

**Review by laurapacilio — APPROVED** [review]
Thank you!! I'm going to merge this once @crw also approves :-)

**Comment by crw** [comment]
@laurapacilio apologies! I think this is good to go.

**Pull request merged**
